vcd/resource_vcd_vapp_network.go

```diff
@@ -220,8 +265,14 @@ func resourceVappNetworkRead(d *schema.ResourceData, meta interface{}) error {
 
 	vAppNetwork := types.VAppNetworkConfiguration{}
 	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
-		if networkConfig.NetworkName == d.Get("name").(string) {
+		networkId, err := govcd.GetUuidFromHref(networkConfig.Link.HREF)
+		if err != nil {
+			return fmt.Errorf("unable to get network ID from HREF: %s", err)
+		}
+		// name check needed to support old resource Id's which was names
```

Comment:
```suggestion
		// Name check needed to support old resource ID's which were names
```

Reply:
changed

Reply:
The new ending is not clear:

> // Name check needed to support old resource ID's which were names **or datasource has only name**

Did you mean this?

> // Name check needed to support old resource IDs which were names **and datasources which use only names for IDs**

But is data source actual here at all? I mean, we're in resource implementation here.

Reply:
comment improved.
We usually reuse resource reads for datasource and make a few simple conditions to handle datasource needs.